### PR TITLE
test(gov/staker): improve test coverage with comprehensive unit tests 

### PR DESCRIPTION
**Pull request by junghoon-vans**
## Summary
- Add comprehensive test coverage for `r/gnoswap/gov/staker` package and its v1 implementation
- Add boundary value tests and edge case coverage for all major components

### New Test Files (Base Package)
- `counter_test.gno`: Counter functionality tests
- `delegation_manager_test.gno`: Delegation manager operations
- `delegation_record_test.gno`: Delegation record handling
- `delegation_snapshot_test.gno`: Snapshot creation and management
- `delegation_withdraw_test.gno`: Withdrawal operations and edge cases
- `emission_reward_state_test.gno`: Emission reward state management
- `launchpad_project_deposits_test.gno`: Launchpad deposit tracking
- `protocol_fee_reward_manager_test.gno`: Protocol fee distribution
- `protocol_fee_reward_state_test.gno`: Protocol fee state management
- `state_test.gno`: State initialization and management
- `store_test.gno`: Store operations
- `upgrade_test.gno`: Upgrade pattern tests

### Enhanced Test Files (v1 Package)
- `api_delegation_test.gno`: API delegation endpoints (200+ lines)
- `api_staker_test.gno`: API staker endpoints (200+ lines)
- `delegation_history_test.gno`: History tracking tests
- `delegation_manager_test.gno`: Enhanced with error cases (404+ lines)
- `delegation_snapshot_test.gno`: Comprehensive snapshot tests (252+ lines)
- `delegation_test.gno`: Core delegation logic (229+ lines)
- `delegation_withdraw_test.gno`: Withdrawal edge cases (128+ lines)
- `emission_reward_manager_test.gno`: Refactored with table-driven tests
- `init_test.gno`: Initialization tests
- `launchpad_project_deposits_test.gno`: Extensive deposit tests (466+ lines)
- `protocol_fee_reward_manager_test.gno`: Refactored with table-driven tests
- `protocol_fee_reward_state_test.gno`: State management tests
- `staker_delegate_test.gno`: Delegation tests
- `util_test.gno`: Utility function tests (382+ lines)

### CODE REVIEW

File: contract/r/gnoswap/gov/staker/v1/delegation_test.gno [hunk 1]
```diff
@@ -695,3 +696,239 @@ func TestDelegation_EdgeCases(t *testing.T) {
...
+			} else {
+				uassert.Nil(t, err)
+				uassert.Equal(t, result, tc.expectedResult)
+			}
+		})
+	}
+}
+
+// TestDelegation_min tests the private min function
+func TestDelegation_min(t *testing.T) {
```
Comment (notJoon) [inline contract/r/gnoswap/gov/staker/v1/delegation_test.gno]: The `min` function only compares integer values using compare operators internally, so this test ends up testing the language specification itself. I think, this appears to be an unnecessary test. how do you think?
